user_immutable/models/res_groups.py

```diff
@@ -0,0 +1,30 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 LasLabs Inc.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+from odoo import api, models
+from odoo.exceptions import AccessError
+
+IMMUTABLE = 'user_immutable.group_immutable'
+
+
+class ResGroups(models.Model):
+
+    _inherit = 'res.groups'
+
+    @api.multi
+    def write(self, vals):
+        """ Override write to verify that access to the `Immutable` group is
+        not given or removed by users without access """
+        immutable_id = -1
+        try:
+            immutable_id = self.env.ref(IMMUTABLE).id
+        except ValueError:
+            pass
```

Comment:
["Errors should never pass silently.](https://www.python.org/dev/peps/pep-0020/)